### PR TITLE
Plugin system testing

### PR DESCRIPTION
**Pull request by br-lewis**
This PR adds some basic tests for loading plugins.

### CODE REVIEW

File: pkg/plugin/plugin_test.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	logger, _ := test.NewNullLogger()
+
+	m := plugin.Manager{
+		Fs:     roFs(),
+		Dir:    subcommandsDir,
+		Logger: logger,
+	}
+
+	plugins := m.Plugins()
+	require.NotEmpty(t, plugins)
```
Comment (bamarni) [inline pkg/plugin/plugin_test.go]: `assert.Equal(t, 1, len(plugins))`

File: pkg/plugin/plugin_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
```
Comment (bamarni) [inline pkg/plugin/plugin_test.go]: `assert.Equal(t, 1, len(plugins))`

File: pkg/plugin/plugin_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	assert.Equal(t, 2, len(plugins))
+}
+
+func targetSubcommandDir(t *testing.T, name string) string {
+	wd, err := os.Getwd()
+	require.NoError(t, err)
+
+	var testdataDir string
+	if runtime.GOOS == "windows" {
+		testdataDir = filepath.Join(wd, "testdata_windows")
```
Comment (bamarni) [inline pkg/plugin/plugin_test.go]: `testdata` is a special convention (https://golang.org/cmd/go/#hdr-Test_packages) : 

> The go tool will ignore a directory named "testdata", making it available to hold ancillary data needed by the tests.

Could you add the `_windows` suffixes to the fixtures folders inside `testdata` instead?

File: pkg/plugin/manager.go [hunk 4]
```diff
@@ -188,14 +188,14 @@ func (m *Manager) loadPluginFromPackage(plugin *Plugin) error {
 func (m *Manager) persist(plugin *Plugin) error {
 	// We need an env directory for `plugin.yaml`.
 	envFilePath := filepath.Join(plugin.dir, "env")
-	if _, err := os.Stat(envFilePath); os.IsNotExist(err) {
+	if _, err := m.Fs.Stat(envFilePath); os.IsNotExist(err) {
```
Comment (bamarni) [inline pkg/plugin/manager.go]: :+1:

File: pkg/plugin/plugin_test.go (hunk 3 above)
Comment (br-lewis) [reply pkg/plugin/plugin_test.go]: How about `testdata/windows/` and `testdata/nix/` instead? I feel like mixing windows and non would make things a bit cluttered.

Comment (bamarni) [reply pkg/plugin/plugin_test.go]: Why `nix`? `unix` is more commonly used in that case (eg. https://github.com/golang/go/tree/master/src/path/filepath).

Comment (br-lewis) [reply pkg/plugin/plugin_test.go]: I've usually seen *nix to refer to all Unix-like OSs but I'm not very attached to that.

**Pull request merged**